plugins/woocommerce/src/Admin/API/Reports/Orders/DataStore.php

```diff
@@ -19,6 +19,14 @@
  */
 class DataStore extends ReportsDataStore implements DataStoreInterface {
 
+	/**
+	 * Dynamically sets the date column name based on configuration
+	 */
+	public function __construct() {
+		$this->date_column_name = get_option( 'woocommerce_date_type', 'date_created' );
+		parent::__construct();
```

Comment:
Is there a reason why `date_column_name` is set before calling the parent constructor? I don't see this as an issue, just curious, as I usually prefer to call the parent constructor immediately unless there is a reason to wait.

Reply:
Yes. Because the parent constructor calls `assign_report_columns` and it was using the wrong value before I had the chance of reading the option